src/mcp/client/session.py

```diff
@@ -59,8 +58,8 @@ class ClientSession(
 ):
     def __init__(
         self,
-        read_stream: MemoryObjectReceiveStream[types.JSONRPCMessage | Exception],
-        write_stream: MemoryObjectSendStream[types.JSONRPCMessage],
+        read_stream: ReadStream,
+        write_stream: WriteStream,
```

Comment:
This is not really the end type, since ReadStream and WriteStream are generics, something is missing here.

I believe it's easier to understand the code if we don't hide that those streams are anyio objects. Makes it easier to go around the code.